test/extensions/filters/listener/original_src/original_src_fuzz_test.cc

```diff
@@ -0,0 +1,50 @@
+#include "envoy/common/exception.h"
+#include "envoy/extensions/filters/listener/original_src/v3/original_src.pb.h"
+
+#include "common/network/utility.h"
+
+#include "extensions/filters/listener/original_src/original_src.h"
+
+#include "test/extensions/filters/listener/original_src/original_src_fuzz_test.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace OriginalSrc {
+
+using testing::_;
+
+DEFINE_PROTO_FUZZER(
+    const envoy::extensions::filters::listener::original_src::OriginalSrcTestCase& input) {
+
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  }
+
+  envoy::extensions::filters::listener::original_src::v3::OriginalSrc proto_config = input.config();
+  Config config(proto_config);
+  auto filter = std::make_unique<OriginalSrcFilter>(config);
```

Comment:
There's an extra copy in here that you don't need. The `=` on L32 is going to copy the contents of input.config() into proto_config, which is going to be passed by const ref to the constructor for `Config`. Consider either inlining `proto_config` or making it a const reference to avoid the copy.

Reply:
I decided to inline it and use input.config() directly instead of copying it into proto_config.